src/m365/aad/commands/app/app-open.ts

```diff
@@ -0,0 +1,123 @@
+
+  public get name(): string {
+    return commands.APP_OPEN;
+  }
+
+  public get description(): string {
+    return 'Opens Azure AD app in the Azure AD portal';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.preview = typeof args.options.preview !== 'undefined';
+    telemetryProps.autoOpenBrowser = typeof args.options.autoOpenBrowser !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .checkAppRegistrationExists(args, logger)
+      .then(_ => this.logOrOpenUrl(args, logger))
+      .then(_ => cb(), (rawRes: any): void => this.handleRejectedODataJsonPromise(rawRes, logger, cb));
+  }
+
+  private checkAppRegistrationExists(args: CommandArgs, logger: Logger): Promise<void> {    
+    const { appId } = args.options;
+
+    if (this.verbose) {
+      logger.logToStderr(`Check if Azure AD app ${appId} exists...`);
+    }
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/myorganization/applications?$filter=appId eq '${encodeURIComponent(appId)}'&$select=id`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { id: string }[] }>(requestOptions)
+      .then((res: { value: { id: string }[] }): Promise<void> => {
+        if (res.value.length === 1) {
+          return Promise.resolve();
+        }
+
+        return Promise.reject(`No Azure AD application registration with ID ${appId} found`);
+      });
+  }
+
+  private logOrOpenUrl(args: CommandArgs, logger: Logger): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const { appId } = args.options;
+      const previewPrefix = args.options.preview === true ? "preview." : "";
+      const url = `https://${previewPrefix}portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/${appId}/isMSAApp/`;
+
+      if (!args.options.autoOpenBrowser) {
+        logger.log(`Use a web browser to open the page ${url}`);
+        return resolve();
+      } 
+      
+      logger.log(`Opening the following page in your browser: ${url}`);
+      
+      // _open is never set before hitting this line, but this check
+      // is implemented so that we stub it when testing.
+      /* c8 ignore next 3 */
+      if (!this._open) {
+        this._open = require('open');
+      }
+
+      (this._open as typeof open)(url).then(() => {
+        resolve();
+      }, (error) => {
+        reject(error);
+      });      
+    });
+    
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      { option: '--appId <appId>' },
+      { option: '--preview [preview]' },
+      { option: '--autoOpenBrowser [autoOpenBrowser]' }
```

Comment:
`--autoOpenBrowser` is meant to be used as a flag so we shouldn't include `[autoOpenBrowser]`